pkg/clusteragent/admission/mutate/autoinstrumentation/injector.go

```diff
@@ -150,9 +150,16 @@ func (i *injector) requirements() libRequirement {
 
 type injectorOption func(*injector)
 
-var injectorVersionAnnotationExtractor = annotationExtractor[injectorOption]{
-	key: "admission.datadoghq.com/apm-inject.version",
-	do:  infallibleFn(injectorWithImageTag),
+var injectorVersionAnnotationExtractorFunc = func(imageResolver ImageResolver) annotationExtractor[injectorOption] {
+	injectorVersionAnnotationExtractor := annotationExtractor[injectorOption]{
```

Comment:
nit: indentation/can just return the annotationExtractor without an intermediate value